src/Storages/MergeTree/ReplicatedMergeTreeSink.cpp

```diff
@@ -79,6 +171,31 @@ struct ReplicatedMergeTreeSinkImpl<async_insert>::DelayedChunk
     std::vector<Partition> partitions;
 };
 
+std::vector<Int64> testSelfDeduplicate(std::vector<Int64> data, std::vector<size_t> offsets, std::vector<String> hashes)
```

Comment:
Move this function to test file? At least to anon namespace?

Reply:
I have no idea on how to deal with it. I wrote it here because in test files we cannot use type `ReplicatedMergeTreeSinkImpl<true>::DelayedChunk::Partition`, which defines in cpp files.

Do we have some MACRO only valid in unit test so that we can avoid compiling it in normal cases?

Reply:
Unfortunately no :(